Example/Example/Sources/Presentation/ViewController.swift

```diff
@@ -83,10 +147,28 @@ class ViewController: UIViewController {
     
     @IBAction private func cancel() {
         progress?.cancel()
+        cancelTask?()
     }
     
+    @MainActor
     private func show(books: [Book]) {
         resultLabel.text = books.map { "• \($0.title)" }.joined(separator: "\n")
     }
+    
+    @MainActor
+    private func show(error: Error) {
+        resultLabel.text = error.localizedDescription
+    }
+    
+    @MainActor
+    private func updateActivityIndicator(isHidden: Bool) {
```

Comment:
It seems that it takes time for extra synchronization and this method could be called from success and failure paths

Reply:
You mean like this?

```swift
let task = detach {
    do {
        let books = try await self.bookService.fetchBooks()
        await self.show(books: books)
        await self.updateActivityIndicator(isHidden: true)
    } catch {
        await self.show(error: error)
        await self.updateActivityIndicator(isHidden: true)
    }
}
```